src/dimension_slice.c

```diff
@@ -354,10 +354,10 @@ dimension_slice_insert_relation(Relation rel, DimensionSlice *slice)
 
 	memset(values, 0, sizeof(values));
 	slice->fd.id = catalog_table_next_seq_id(catalog_get(), DIMENSION_SLICE);
-	values[Anum_dimension_slice_id - 1] = slice->fd.id;
```

Comment:
just for reference, what are these if you don't cast them?

Reply:
also, are there other places we need to check and explicitly cast?

Reply:
So if you don't pass it through *GetDatum then this is just a memory copy. This works if the memory representation of what you are setting is the same as what the Datum uses. However, this is architecture dependent. I tried to find all similar places.